### PR TITLE
[RNMobile] Media & Text - Media picker buttons functionality

### PR DESCRIPTION
**Pull request by geriux**
Can be merged after https://github.com/WordPress/gutenberg/pull/17392

`gutenberg-mobile` PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/1378

## Description
This PR adds the missing functionality for the Media & Text block https://github.com/wordpress-mobile/gutenberg-mobile/issues/1313

On **Android** It will only support images until we can filter by more than one type at a time (images & videos) in the media picker.

Now the functionality it's the same as the `image` and `video` blocks.

## How has this been tested?

- Open the WordPress app and start a new post.
- Insert a new `Media & Text` block.
- Add an image from the photo library/camera.
- Expect the progress bar to show up while the file is being uploaded. While it's doing that you can cancel the upload.
- Once an image is uploaded you can do a long press to open the media options to change the file.
- If the connection fails, a retry option will be available.
- Test the image/video can be also selected from the WordPress media library.

### iOS

    

    

    

---

### Android

    

    

---

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows the accessibility standards.
- [x] My code has proper inline documentation.
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/block-library/src/media-text/media-container.native.js
Comment (koke) [inline packages/block-library/src/media-text/media-container.native.js]: I think you can set the [aspectRatio](https://facebook.github.io/react-native/docs/layout-props#aspectratio) style prop directly and avoid the calculation. Otherwise, I think using the window width would be problematic because the ratio should apply to the container width, not the full window, but I haven't been able to test this because of the other issue I commented

File: packages/block-library/src/media-text/media-container.native.js
Comment (koke) [inline packages/block-library/src/media-text/media-container.native.js]: I don't love that we are hardcoding this, but I see we were already doing it in the video block. Can we at least move that logic to the newly extracted VideoPlayer component so we can improve it in the future?

File: packages/block-library/src/media-text/media-container.native.js
Comment (koke) [inline packages/block-library/src/media-text/media-container.native.js]: I'd suggest [Platform.select](https://facebook.github.io/react-native/docs/platform-specific-code#platform-module) for a bit more clarity:

```suggestion
const ALLOWED_MEDIA_TYPES = Platform.select({
	android: [ MEDIA_TYPE_IMAGE ],
	default: [ MEDIA_TYPE_IMAGE, MEDIA_TYPE_VIDEO ],
});
```

File: packages/block-library/src/media-text/media-container.native.js
Comment (koke) [reply packages/block-library/src/media-text/media-container.native.js]: Confirmed that this gets the wrong aspect ratio when uploading a video on iPad:

IMG_D8955C916037-1

File: packages/block-library/src/media-text/media-container.native.js
Comment (geriux) [reply packages/block-library/src/media-text/media-container.native.js]: Oh yes! Will do!

File: packages/block-library/src/media-text/media-container.native.js
Comment (geriux) [reply packages/block-library/src/media-text/media-container.native.js]: I agree, way better to see! I'll change it

File: packages/block-library/src/media-text/media-container.native.js
Comment (geriux) [reply packages/block-library/src/media-text/media-container.native.js]: Nice! I'll check `aspectRatio` and also make sure it looks great on tablets as well

**Comment by geriux** [comment]
Hey @koke this is ready to be reviewed again, Travis failed due to a linting issue that will be fixed in https://github.com/WordPress/gutenberg/pull/17598

**Comment by koke** [comment]
I think this is looking good. I had something else in mind for the aspect ratio changes, but now I realize that the video component probably isn't as smart as images and we want a fixed ratio there. I'd probably like to have in the future an API for VideoPlayer and the placeholder component that does the right thing by itself, but I think this is good for now.

Let's try to wrap up #17392 first and merge that one and the lint fixes from master before I do the final testing round. 👍

**Comment by geriux** [comment]
Updated! Now all checks are green 🎉 ready to be reviewed again =)

**Comment by koke** [comment]
I noticed one thing that we don't have to fix in this PR but it's worth discussing with @iamthomasbishop. I wanted to change the media once I had uploaded it. When the block is selected, the toolbar is visible, but since there is no focus on any text component, the toolbar is at the bottom:

Simulator Screen Shot - iPhone 11 - 2019-09-30 at 11 55 04

I stared at the screen and tapped on too many things for a minute trying to edit the image/video, but I just couldn't see the toolbar at the bottom.

**Comment by iamthomasbishop** [comment]
@koke I think the toolbar positioning is expected, but I wouldn't expect the `pencil` icon to be visible until I have selected the video block within the group. I know the web currently follows the above pattern, but it's always confusing to me. Shouldn't that button only be shown when I've selected the video block?

**Comment by koke** [comment]
@iamthomasbishop the left part of media & text doesn't use nested blocks, it's just either one image or one video, so it's part of the parent (media-text) block. I understand that was the expected design, but I still found myself confused for a bit trying to find how to replace the image/video there

**Comment by iamthomasbishop** [comment]
> the left part of media & text doesn't use nested blocks, it's just either one image or one video, so it's part of the parent (media-text) block

Yea sorry, I thought from the example above that the _parent_ (media-text) was selected. The replacing mechanism is definitely confusing, I've always disliked that icon. I think the web has recently updated that `replace image` icon (at least on the image block) to be more obvious, can we match that?  Example:

Screen Shot 2019-10-01 at 1 49 21 PM

The other thing that I mentioned still applies – why do we even show the image/media replace button while the _parent_ is selected? That's confusing to me.

**Comment by koke** [comment]
I'm seeing some odd layouts after the merge:

IMG_21A06FC3B1EC-1

I'm not sure what could be causing that but I'm not seeing the same on develop. Note that there are some stacking issues being addressed in #17682, but not the text landing on top of the media.

**Comment by geriux** [comment]
> I'm not sure what could be causing that but I'm not seeing the same on develop. Note that there are some stacking issues being addressed in #17682, but not the text landing on top of the media.

I saw those and thought it was about the stacking issue. It was definitely not happening before rebasing with master. I'll merge #17682 in a test branch to see how it looks.

File: test/native/setup.js
Comment (pinarol) [inline test/native/setup.js]: Why do we need this one for this PR?

Comment (geriux) [reply test/native/setup.js]: Ahh no we don't sorry,  I was using `Platform.select` but ended up removing it when I made some changes. I'll update it 👍

**Comment by koke** [comment]
> I'll merge #17682 in a test branch to see how it look

I tried that myself when testing and it went full width but still had the same issue.

File: test/native/setup.js
Comment (geriux) [reply test/native/setup.js]: Removed =)

**Comment by geriux** [comment]
@koke I've just updated this PR with the fix for the issue you mentioned, I also tested this fix with your PR and it works well too 🎉

**Comment by koke** [comment]
I feel like I had reported this before, but I can't find my comment anywhere so maybe I didn't? 😱 

1. Add a media-text block
2. Tap Add image or video
3. Choose any method
4. Tap cancel without selecting an image
5. Red screen: `Cannot read property 'id' of undefined`

**Comment by geriux** [comment]
> I feel like I had reported this before, but I can't find my comment anywhere so maybe I didn't? 😱
> 
> 1. Add a media-text block
> 2. Tap Add image or video
> 3. Choose any method
> 4. Tap cancel without selecting an image
> 5. Red screen: `Cannot read property 'id' of undefined`

Oh wow! I haven't seen that D: I'll check it out today. Did the other issue get fixed with the latest changes?

**Comment by koke** [comment]
I think this is good to go once the [red screen](https://github.com/WordPress/gutenberg/pull/17537#issuecomment-538969695) issue is solved. 🎉

**Review by koke — APPROVED** [review]
I think this is ready 🎉 
To avoid further issues, I'll wait until https://github.com/wordpress-mobile/gutenberg-mobile/pull/1410 is merged to start merging all the related PRs

**Review by mkevins — APPROVED** [review]
I've tested this on Android with the additional changes here: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1412 and here: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1400. This is looking good. Nice work @geriux ! :tada:

**Pull request merged**
